### PR TITLE
Personal favs refactor

### PR DESCRIPTION
**Pull request by Dima-1**

### CODE REVIEW

File: OsmAnd/src/net/osmand/data/FavouritePoint.java [hunk 1]
```diff
@@ -21,6 +25,51 @@
 	private int color;
 	private boolean visible = true;
 
+	public enum PointType {
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/data/FavouritePoint.java]: This class probably not needed

File: OsmAnd/src/net/osmand/data/FavouritePoint.java [hunk 2]
```diff
@@ -21,6 +25,51 @@
...
+		HOME("home", R.string.home_button, 1, R.drawable.ic_action_home_dark),
+		WORK("work", R.string.work_button, 2, R.drawable.ic_action_work),
+		PARKING("parking", R.string.map_widget_parking, 3, R.drawable.ic_action_parking_dark);
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/data/FavouritePoint.java]: You can't use "map_" or "widget_" icons in UI they have different size for Tablet

File: OsmAnd/src/net/osmand/data/FavouritePoint.java [hunk 3]
```diff
@@ -101,12 +154,23 @@ public void setLongitude(double longitude) {
...
+	}
 	
 	public void setCategory(String category) {
 		this.category = category;
 	}
 
-	public String getName(Context ctx) {
+	public String getName(@NonNull Context ctx) {
+		if (isPersonal()) {
+			return PointType.valueOfTypeName(getName()).getHumanString(ctx);
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/data/FavouritePoint.java]: if(HOME.equals) { return .. .}  else if ... 
will be better for now

File: OsmAnd/src/net/osmand/data/FavouritePoint.java [hunk 4]
```diff
@@ -101,12 +154,23 @@ public void setLongitude(double longitude) {
 	public String getCategory() {
 		return category;
 	}
+
+	public String getCategory(@NonNull Context ctx) {
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/data/FavouritePoint.java]: Method should be called getCategoryDisplayName

File: OsmAnd/src/net/osmand/data/FavouritePoint.java [hunk 5]
```diff
@@ -54,12 +103,16 @@ public PointDescription getPointDescription() {
...
 	public boolean isPersonal() {
-		return false;
+		return PERSONAL_CATEGORY.equals(category);
 	}
 
 	@Override
-	public PointDescription getPointDescription(Context ctx) {
-		return getPointDescription();
+	public PointDescription getPointDescription(@NonNull Context ctx) {
+		if (isPersonal()) {
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/data/FavouritePoint.java]: not clear difference between personal and non-persona. I think it should be identical in both cases return new PointDescription(PointDescription.POINT_TYPE_FAVORITE, getName(ctx));

File: OsmAnd/src/net/osmand/plus/activities/FavoritesTreeFragment.java [hunk 6]
```diff
@@ -752,8 +760,31 @@ void synchronizeGroups() {
 			for (FavoriteGroup key : gs) {
 				boolean empty = true;
 				if (flt == null || flt.contains(key)) {
-					empty = false;
-					favoriteGroups.put(key, new ArrayList<>(key.points));
+					if (key.personal) {
+						ArrayList<FavouritePoint> list = new ArrayList<>();
+						for (FavouritePoint p : key.points) {
+							if (p.getName().equals(FavouritePoint.PointType.PARKING.getName())) {
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/activities/FavoritesTreeFragment.java]: 1. it is wrong place to customize "isParkingPluginEnable" should not be here in the code
2. there is no need fofr this ifs , it should be always visible 
Code should be deleted

File: OsmAnd/src/net/osmand/data/FavouritePoint.java (hunk 2 above)
Comment (Dima-1) [reply OsmAnd/src/net/osmand/data/FavouritePoint.java]: I use only "ic_action_" icons

File: OsmAnd/src/net/osmand/plus/activities/FavoritesTreeFragment.java (hunk 6 above)
Comment (Dima-1) [reply OsmAnd/src/net/osmand/plus/activities/FavoritesTreeFragment.java]: What if parking plugin will be disabled? We must hide parking point?

File: OsmAnd/src/net/osmand/plus/FavouritesDbHelper.java [hunk 7]
```diff
@@ -75,6 +73,14 @@ public FavouritesDbHelper(OsmandApplication context) {
 		public int color;
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/FavouritesDbHelper.java]: get/set

File: OsmAnd/src/net/osmand/plus/FavouritesDbHelper.java [hunk 8]
```diff
@@ -75,6 +73,14 @@ public FavouritesDbHelper(OsmandApplication context) {
...
 		public boolean personal = false;
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/FavouritesDbHelper.java]: delete this field

File: OsmAnd/src/net/osmand/data/FavouritePoint.java [hunk 9]
```diff
@@ -3,14 +3,18 @@
...
 
 public class FavouritePoint implements Serializable, LocationPoint {
 	private static final long serialVersionUID = 729654300829771466L;
 
-	protected static final String HIDDEN = "hidden";
+	private static final String HIDDEN = "hidden";
+	public static final String PERSONAL_CATEGORY = "personal";
 
 	protected String name = "";
 	protected String description;
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/data/FavouritePoint.java]: protected String address;

File: OsmAnd/src/net/osmand/plus/views/FavouritesLayer.java [hunk 10]
```diff
@@ -133,6 +135,10 @@ public void onPrepareBufferImage(Canvas canvas, RotatedTileBox tileBox, DrawSett
 									continue;
 								}
 							}
+							if (group.personal && o.getName().equals(FavouritePoint.PointType.PARKING.getName())
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/views/FavouritesLayer.java]: bad code

File: OsmAnd/src/net/osmand/plus/views/FavouritesLayer.java [hunk 11]
```diff
@@ -291,6 +299,7 @@ public void applyNewObjectPosition(@NonNull Object o,
 		boolean result = false;
 		if (o instanceof FavouritePoint) {
 			favorites.editFavourite((FavouritePoint) o, position.getLatitude(), position.getLongitude());
+			favorites.lookupAddressAllPersonalPoints();
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/views/FavouritesLayer.java]: ?? All personal points ??

File: OsmAnd/src/net/osmand/plus/views/FavouritesLayer.java [hunk 12]
```diff
@@ -204,7 +210,9 @@ private void getFavFromPoint(RotatedTileBox tb, List<? super FavouritePoint> res
 			int x = (int) tb.getPixXFromLatLon(n.getLatitude(), n.getLongitude());
 			int y = (int) tb.getPixYFromLatLon(n.getLatitude(), n.getLongitude());
 			if (calculateBelongs(ex, ey, x, y, r)) {
-				res.add(n);
+				if (!n.getName().equals(FavouritePoint.PointType.PARKING.getName())) {
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/views/FavouritesLayer.java]: bad code

File: OsmAnd/src/net/osmand/data/FavouritePoint.java [hunk 13]
```diff
@@ -101,12 +154,23 @@ public void setLongitude(double longitude) {
...
+			return category;
+		}
+	}
 	
 	public void setCategory(String category) {
 		this.category = category;
 	}
 
-	public String getName(Context ctx) {
+	public String getName(@NonNull Context ctx) {
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/data/FavouritePoint.java]: getDisplayName

File: OsmAnd/src/net/osmand/plus/FavouritesDbHelper.java [hunk 14]
```diff
@@ -744,8 +750,8 @@ public int compare(FavoriteGroup lhs, FavoriteGroup rhs) {
 			@Override
 			public int compare(FavouritePoint o1, FavouritePoint o2) {
 				if (o1.isPersonal() && o2.isPersonal()) {
-					int x = ((PersonalFavouritePoint) o1).getType().getOrder();
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/FavouritesDbHelper.java]: ?

**Pull request merged**